### PR TITLE
impl backend server. 

### PR DESCRIPTION
**Pull request by yangzhi-tech**
TODO 1.Use mongodb. 2.Send email. 3. Update orde…r status 4.order management

### CODE REVIEW

**Review by BraisedSix — COMMENTED** [review]
Code reviewer: wenxing, review results, no unreasonable code was found, the code is very OK

File: server/payment/client.go [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+	if err != nil {
+		return nil, err
+	}
+	defer res.Body.Close()
+	if res.StatusCode != 200 {
+		msg, err := io.ReadAll(res.Body)
+		if err != nil {
+			return nil, err
+		}
+		return nil, fmt.Errorf("Retcode %d, msg %x", res.StatusCode, msg)
```
Comment (tim-nju) [inline server/payment/client.go]: msg %x? Should it be %s or has it particular intention?

File: server/store/mem_store.go [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+// For test purpose
+type MemStore struct {
+	orders map[string]*model.Order
+	lock   sync.Mutex
+}
+
+func (s *MemStore) CreateOrder(o *model.Order) error {
+	s.lock.Lock()
+	defer s.lock.Unlock()
+	s.orders[o.PaymentID] = o
```
Comment (tim-nju) [inline server/store/mem_store.go:23]: Should we consider error case like the order id has existed already?

Comment (yangzhi-tech) [reply server/store/mem_store.go:23]: I guess nowpayment will always generate new paymentid.
But this is a good point. Will check this in the next PR.

File: server/payment/client.go (hunk 1 above)
Comment (yangzhi2105) [reply server/payment/client.go:75]: %x for print bytes.

Comment (yangzhi-tech) [reply server/payment/client.go]: Good catch! This is a typo. Will fix in next commit.

**Review by tim-nju — APPROVED** [review]

File: server/store/mongo_store.go [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+const orderCollection = "order"
+
+func NewMongoStore(cstr string) *MongoStore {
+	if cstr == "" {
+		os.Exit(1)
+	}
+	return &MongoStore{cstr: cstr}
+}
+
+func (s *MongoStore) Connect() error {
```
Comment (tim-nju) [inline server/store/mongo_store.go:27]: I am not sure but do we also need Disconnnect?

File: server/store/mongo_store.go [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+	}
+	s.client = c
+	err = c.Connect(context.Background())
+	if err != nil {
+		return err
+	}
+	return c.Ping(context.Background(), nil)
+}
+
+func (s *MongoStore) CreateOrder(order *model.Order) error {
```
Comment (tim-nju) [inline server/store/mongo_store.go:41]: I am not sure whether it is necessary to pass in the context to allow for greater control over request timeouts and cancellations

Comment (yangzhi-tech) [reply server/store/mongo_store.go:41]: updated

**Pull request merged**